.evergreen/csfle/README.md

```diff
@@ -0,0 +1,80 @@
+# Scripts for In-Use Encryption testing
+
+This folder contains scripts for use with In-Use Encryption.
+
+See the [Azure KMS](./azurekms/README.md) and [GCP KMS](./gcpkms/README.md)
+for more information on those specific scenarios.
+
+## Prerequisities
+
+The system you are running on must have Python 3 and have access to the
+`drivers/csfle` [AWS Vault](https://wiki.corp.mongodb.com/display/DRIVERS/Using+AWS+Secrets+Manager+to+Store+Testing+Secrets).
+For legacy usage, see below.
+
+## Usage
+
+Set up In-Use Encryption by first fetching the secrets and then launching the kms servers:
+
+```bash
+$DRIVERS_TOOLS/.evergreen/csfle/setup_secrets.sh
+$DRIVERS_TOOLS/.evergreen/csfle/start_servers.sh
+```
+
+The generated `secrets-export.sh` file can be sourced from your `cwd` to get the required credentials for testing.
+
+The following servers will be started:
+
+- Mock KMIP server on port 5698
+- KMS HTTP server with an expired cert on port 8000
+- KMS HTTP server with an "wrong host" cert on port 8001
+- KMS HTTP server with a correct cert on port 8002
+- Mock Azure IMDS server on port 8080
+
+When finished, stop the servers by running:
+
+```bash
+$DRIVERS_TOOLS/.evergreen/csfle/stop_servers.sh
+```
+
+If you are starting your CSFLE servers in a separate Evergreen task, it is recommended that you setup secrets
```

Comment:
Since starting the servers does not block, I expect a separate Evergreen task is not needed. I expect `await_servers.sh` can be combined into `start_servers.sh`. I expect the sample Evergreen config can be simplified to use one `subprocess.exec`. Example (untested):

```yml
start-csfle-servers:
- command: subprocess.exec
    params:
    working_dir: src
    binary: bash
    background: true
    include_expansions_in_env: ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_SESSION_TOKEN", "DRIVERS_TOOLS"]
    args:
        - ./scripts/setup-encryption.sh
- command: subprocess.exec
    params:
    working_dir: src
    binary: bash
    args:
      - |
        # Whatever other setup needed here, like setting CSFLE_TLS_CA_FILE
        bash ${DRIVERS_TOOLS}/.evergreen/csfle/setup_secrets.sh
        bash ${DRIVERS_TOOLS}/.evergreen/csfle/start_and_await_servers.sh
```

Reply:
I had tried doing this, but if it isn't run in the background, you end up with:

```
[2024/01/24 13:33:03.250] Waiting on KMIP server on port 5698...done
[2024/01/24 14:27:11.488] Hit exec timeout (1h0m0s).
```

It does not happen if the function is run within the same task as the test or run in the background.  I believe this is because the EVG runner is waiting for all of the forked processes to stop.

Reply:
Ah I see. Sorry for the incorrect suggestion.